lib/bank_accounts/account.rb

```diff
@@ -0,0 +1,35 @@
+module BankAccounts
+  class Account
+    attr_reader :id, :balance
+
+    def initialize(id, initial_balance)
+      @id = id
+
+      # must have at least $0 balance
+      if initial_balance < 0
+        raise ArgumentError, "Account can't have a negative balance. Create a new account."
+      else
+        @balance = initial_balance
```

Comment:
Smart use of `@balance` as your instance variable and then using the `attr_reader` so you didn't have to write a separate method